File: jambeth/jAmbeth.Util/src/de/osthus/ambeth/util/TypeUtil.java
package de.osthus.ambeth.util;

import de.osthus.ambeth.repackaged.org.objectweb.asm.Type;

public final class TypeUtil
{
	public static final Type[] getClassesToTypes(Class<?>[] classes)
	{
		Type[] types = new Type[classes.length];
		for (int a = classes.length; a-- > 0;)
		{
			Class<?> clazz = classes[a];
			if (clazz == null)
			{
				continue;
			}
			types[a] = Type.getType(clazz);
		}
		return types;
	}

	private TypeUtil()
	{
		// Intended blank
	}
}

File: ambeth/Ambeth.Util/ambeth/util/TypeUtil.cs
using System;
using System.Reflection;

namespace De.Osthus.Ambeth.Util
{
    public sealed class TypeUtil
    {
        public static NewType[] GetClassesToTypes(Type[] classes)
        {
            NewType[] types = new NewType[classes.Length];
            for (int a = classes.Length; a-- > 0; )
            {
                Type clazz = classes[a];
                if (clazz == null)
                {
                    continue;
                }
                types[a] = NewType.GetType(clazz);
            }
            return types;
        }

        public static Type[] GetParameterTypesToTypes(ParameterInfo[] classes)
        {
            Type[] types = new Type[classes.Length];
            for (int a = classes.Length; a-- > 0; )
            {
                types[a] = classes[a].ParameterType;
            }
            return types;
        }

        public static NewType[] GetClassesToTypes(ParameterInfo[] classes)
        {
            NewType[] types = new NewType[classes.Length];
            for (int a = classes.Length; a-- > 0; )
            {
                types[a] = NewType.GetType(classes[a].ParameterType);
            }
            return types;
        }

        public static Type[] GetClassesToTypesNative(ParameterInfo[] classes)
        {
            Type[] types = new Type[classes.Length];
            for (int a = classes.Length; a-- > 0; )
            {
                types[a] = classes[a].ParameterType;
            }
            return types;
        }

        private TypeUtil()
        {
            // Intended blank
        }
    }
}